src/app/components/calendar/calendar.component.ts

```diff
@@ -22,9 +60,25 @@ export class CalendarComponent {
 		select: (selectedRange) => this.selectDates(selectedRange),
 	};
 
-	selectDates(selectedRange: DateSelectArg): void {
-		const event = { title: "eventName", start: selectedRange.start, end: selectedRange.end };
+	private selectDates(selectedRange: DateSelectArg): void {
+		const event: CalendarEvent = { title: "eventName", start: selectedRange.start, end: selectedRange.end };
 		this.events = [...this.events, event];
 		this.calendarOptions.events = this.events;
 	}
+
+	public getInitials() {
+		if (!this.userData || !this.userData.firstName || !this.userData.lastName) {
```

Comment:
add in this method a console.log("getInitials()") and open Web Console. Count how many of getInitials() calls are in this component. 
This is a performance issue. The best approach is "never call a method from ts directly, create instead a pure pipe and move the logic there". If you need hints, please let me know